### PR TITLE
DST-182: clean up ingestion functions

### PR DESCRIPTION
**Pull request by ccheng26**
## Ticket

Resolves DST-182
by comparing the chunk size / overlap with the embedding function supplied to the vectordb argument and raise a warning there if they don't match

## Changes
- updated `ingest.py` file with code from `optimize_encoder.py`
- added defaults for tools

## Testing Tools

Testing instructions and expected behavior:
- household eligibility bot
1. run ./chainlit-household-bot.py 
2. select all-MiniLM-L6-v2 (the hugging face embedding fn seems to be broken, unrelated to this PR)for embedding option and other llm besides google
3. upload a pdf and/or json data
4. ingestion should succeeed
5. delete chromadb/

Note `python run.py` already has `print(f"Max sequence length (max tokens):` so we're not passing the embedding info
1. run python run.py
2. select any llm
3. select any option with ingestion

- run python optimize_encoder.py
- run python eval.py

### CODE REVIEW

File: 02-household-queries/chainlit-household-bot.py [hunk 1]
```diff
@@ -348,11 +357,18 @@ async def on_click_upload_file_query(action: cl.Action):
         # initialize db
         await set_vector_db()
         vectordb = cl.user_session.get("vectordb")
+        settings = cl.user_session.get("settings")
+        embeddings = settings["embedding"]
```
Comment (yoomlam) [inline 02-household-queries/chainlit-household-bot.py:361]:
```suggestion
        embedding = settings["embedding"]
```

File: 02-household-queries/chainlit-household-bot.py [hunk 2]
```diff
@@ -348,11 +357,18 @@ async def on_click_upload_file_query(action: cl.Action):
...
+
         if file.type == "application/pdf":
-            add_pdf_to_vector_db(vectordb=vectordb, file_path=file.path)
+            add_pdf_to_vector_db(
+                vectordb=vectordb, file_path=file.path, embedding_name=embeddings
```
Comment (yoomlam) [inline 02-household-queries/chainlit-household-bot.py:365]:
```suggestion
                vectordb=vectordb, file_path=file.path, embedding_name=embedding
```

File: 02-household-queries/chainlit-household-bot.py [hunk 3]
```diff
@@ -348,11 +357,18 @@ async def on_click_upload_file_query(action: cl.Action):
...
+            )
         elif file.type == "application/json":
             add_json_html_data_to_vector_db(
                 vectordb=vectordb,
+                embedding_name=embeddings,
```
Comment (yoomlam) [inline 02-household-queries/chainlit-household-bot.py:370]:
```suggestion
                embedding_name=embedding,
```

File: 02-household-queries/ingest.py [hunk 4]
```diff
@@ -1,33 +1,107 @@
...
         print("  Split into", len(texts))
+    for t in texts:
+        token_count = llm.get_num_tokens(t)
+        overlap_and_token = token_count + chunk_overlap
+        if token_count > token_limit:
+            print(f"Exceeded token limit of {token_limit}: {token_count};")
+        elif chunk_size > (overlap_and_token):
+            print(
+                f"Exceeded token count and overlap {overlap_and_token}: {chunk_size};"
+            )
```
Comment (yoomlam) [inline 02-household-queries/ingest.py:85]: - `chunk_size` and `chunk_overlap` are measured in number of characters
- `token_count` is measured in number of tokens/floats
What is the unit of measure for `overlap_and_token`? The number of tokens (for `token_count`) is not comparable to the number of characters (for `chunk_overlap`).

What is the reasoning behind this `elif` block?

File: 02-household-queries/ingest.py [hunk 5]
```diff
@@ -1,33 +1,107 @@
...
     entire_text = source + "\n\n" + text
     texts = text_splitter.split_text(entire_text)
+
     if not silent:
         print("  Split into", len(texts))
+    for t in texts:
+        token_count = llm.get_num_tokens(t)
+        overlap_and_token = token_count + chunk_overlap
+        if token_count > token_limit:
+            print(f"Exceeded token limit of {token_limit}: {token_count};")
```
Comment (yoomlam) [inline 02-household-queries/ingest.py:81]: Crux of the PR

File: 02-household-queries/chainlit-household-bot.py [hunk 6]
```diff
@@ -328,7 +335,9 @@ async def on_click_upload_default_files(action: cl.Action):
     msg = cl.Message(content="Processing files...", disable_feedback=True)
     await msg.send()
 
-    ingest_call(vectordb)
+    settings = cl.user_session.get("settings")
+    embeddings = settings["embedding"]
+    ingest_call(vectordb, embeddings)
```
Comment (yoomlam) [inline 02-household-queries/chainlit-household-bot.py:340]: Use consistent pluralization
```suggestion
    embedding = settings["embedding"]
    ingest_call(vectordb, embedding)
```

**Review by yoomlam — CHANGES_REQUESTED** [review]

File: 02-household-queries/ingest.py (hunk 4 above)
Comment (ccheng26) [reply 02-household-queries/ingest.py:85]: I was under the assumption the what was in the processed would be determined by the chunk size, overlap and the base token amount given the text. I'll remove this

**Review by yoomlam — APPROVED** [review]
Code LGTM! Untested

**Pull request merged**